src/services/external-api-helper.ts

```diff
@@ -198,33 +194,187 @@ export const getFromIMDbAPIV2 = async(imdbId?: string, searchRequest?: SearchReq
 };
 
 /**
- * Sets and returns series metadata by IMDb ID.
- *
- * @param imdbID the IMDb ID of the series.
+ * @param seriesTitle the series title
+ * @param [year] the year the series started
+ * @returns the TMDB ID of the first TV series from
+ *          the list of results from TMDB by title
  */
-export const setSeriesMetadataByIMDbID = async(imdbID: string): Promise<SeriesMetadataInterface | null> => {
-  if (!imdbID) {
-    throw new Error('IMDb ID not supplied');
+const getSeriesTMDBIDFromTMDBAPI = async(seriesTitle: string, year?: number): Promise<number> => {
+  const tmdbQuery: SearchTvRequest = { query: seriesTitle };
+  if (year) {
+    // eslint-disable-next-line @typescript-eslint/camelcase
+    tmdbQuery.first_air_date_year = year;
+  }
+  const searchResults = await moviedb.searchTv(tmdbQuery);
+  if (searchResults?.results && searchResults.results[0] && searchResults.results[0].id) {
+    return searchResults.results[0].id;
   }
 
-  // Shouldn't really happen since we got this IMDb ID from their API
-  if (await FailedLookups.findOne({ imdbID }, '_id', { lean: true }).exec()) {
-    await FailedLookups.updateOne({ imdbID }, { $inc: { count: 1 } }).exec();
-    return null;
+  return null;
+};
+
+/**
+ * Attempts a query to the TMDB API and standardizes the response
+ * before returning.
+ *
+ * @param [movieOrSeriesTitle] the title of the movie or series
+ * @param [imdbID] the IMDb ID of the movie or series
+ * @param [year] the year of first release
+ * @param [seasonNumber] the season number if this is an episode
+ * @param [episodeNumber] the episode number if this is an episode
+ */
+export const getFromTMDBAPI = async(movieOrSeriesTitle?: string, imdbID?: string, year?: number, seasonNumber?: number, episodeNumber?: number): Promise<MediaMetadataInterface | null> => {
+  if (!movieOrSeriesTitle && !imdbID) {
+    throw new Error('Either movieOrSeriesTitle or IMDb ID must be specified');
   }
+  // If the client specified an episode number, this is an episode
+  const isExpectingTVEpisode = Boolean(episodeNumber);
+
+  let metadata;
+  if (isExpectingTVEpisode) {
+    // TODO: Handle imdbID for episodes
```

Comment:
Was this a todo for now or later?

Reply:
Hmm I didn't think it would really ever get reached, but after looking again I guess OpenSubtitles could return that ID so I should do that now